coursera/downloaders.py

```diff
@@ -363,6 +364,17 @@ def _start_download(self, url, filename, resume=False):
             chunk_sz = 1048576
             progress = DownloadProgress(content_length)
             progress.start()
+            
```

Comment:
Hi.

Sorry for the late review, but I see that you are only changing the name with the built-in downloader.

I have inspected the code and I made some local modifications here that incorporate your regular expression work, but I have put the code up in the class hierarchy, in the base class `Downloader`, which is inherited both by `ExternalDownloader` and also by `NativeDownloader`.

In fact, I also called the function `clean_filename` from the `util` module and, perhaps, these regular expressions should be put there. Comments?

I also changed the regular expression below to stop the filenames at either a `?` or a `#`, since those thave special meaning only for URL GET verbs. Oh, and this is after URL escaping.

@balta2ar? @iemejia? @Pragith?